File: rubyrays/camera.rb
require_relative 'vector'

class Camera
  G = Vector.new(-3.1, -16, 1.9).norm

  A = (Vector::NORMAL.cross(G)).norm * 0.002
  B = (G.cross(A)).norm * 0.002
  C = (A + B) * -256 + G

  ORIGIN = Vector.new(-5, 16, 8)
end

File: javarays/Camera.java
package javarays;

public final class Camera {
    static final RayVector G = (new RayVector(-3.1f, -16.f, 1.9f)).norm();
    static final RayVector UP = (new RayVector(0.f,  0.f,  1.f).cross(G)).norm().scale(.002f);
    static final RayVector RIGHT = (G.cross(UP)).norm().scale(.002f);
    static final RayVector EYE_OFFSET = (UP.add(RIGHT)).scale(-256).add(G);
    static float aspectRatio;

    // Ray Origin
    public static final RayVector ORIGIN   = new RayVector(-5.f, 16.f,  8.f);
}
